haystack/preview/components/websearch/serper_dev.py

```diff
@@ -0,0 +1,161 @@
+            organic = [
+                Document.from_dict(
+                    {"metadata": {k: v for k, v in d.items() if k != "snippet"}, "content": d["snippet"]}
+                )
+                for d in json_result["organic"]
+            ]
+
+            answer_box = []
+            # answer box is what search engine shows as a direct answer to the query
+            if "answerBox" in json_result:
+                answer_dict = json_result["answerBox"]
+                highlighted_answers = answer_dict.get("snippetHighlighted")
+                answer_box_content = (
+                    highlighted_answers[0]
+                    if isinstance(highlighted_answers, list) and len(highlighted_answers) > 0
+                    else highlighted_answers
+                    if isinstance(highlighted_answers, str)
+                    else None
+                )
+                if not answer_box_content:
+                    for key in ["snippet", "answer", "title"]:
+                        if key in answer_dict:
+                            answer_box_content = answer_dict[key]
+                            break
+                if answer_box_content:
+                    answer_box = [
+                        Document.from_dict(
+                            {
+                                "content": answer_box_content,
+                                "metadata": {
+                                    "title": answer_dict.get("title", ""),
+                                    "link": answer_dict.get("link", ""),
+                                },
+                            }
+                        )
+                    ]
+
+            people_also_ask = []
+            # these are related questions that search engine shows
+            if "peopleAlsoAsk" in json_result:
+                for result in json_result["peopleAlsoAsk"]:
+                    title = result.get("title", "")
+                    people_also_ask.append(
+                        Document.from_dict(
+                            {
+                                "content": result["snippet"] if result.get("snippet") else title,
+                                "metadata": {"title": title, "link": result.get("link", None)},
+                            }
+                        )
+                    )
+
+            documents = answer_box + organic + people_also_ask
+
+            logger.debug("Serper Dev returned %s documents for the query '%s'", len(documents), query)
+            result_docs = documents[:top_k]
+            return score_results(result_docs, len(answer_box) > 0)
+
+        except requests.Timeout:
+            raise TimeoutError(f"Request to {self.__class__.__name__} timed out.")
+
+        except requests.RequestException as e:
+            raise Exception(f"An error occurred while querying {self.__class__.__name__}. Error: {e}")
+
+        except Exception as e:
+            raise Exception(f"An unexpected error occurred: {e}")
```

Comment:
I'd wrap only the actual request with `try`. Also I wouldn't catch `Exception` and raise it again, that's unnecessary.